### PR TITLE
73 implement e2e test

### PR DESCRIPTION
**Pull request by lorerod**
# Description

Implement the e2e test for one first scenario: The LTFU flow happy path

medic/interoperability#[73](https://github.com/medic/cht-interoperability/issues/73)

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by lorerod** [comment]
Hi! @andrablaj @njogz @samuelimoisili! I will appreciate your thought about this.
The way I thought about this solution was to be able to test the exact steps I execute when testing the workflow locally.
Everything begins in mediator/test/e2e-test.sh. The script inits the containers and runs the ltfu workflow calling mediator/test/ltfu-flow.spec.ts. 
I started this spec having one test (it) per step but felt that it wasn't appropriate because every test needs to work independently, which will end up being super slow. So I decided to make one test for the entire workflow.
Right now, the ci pipeline is failing for the e2e. I will check on this first thing on Monday. I suspect it is the base dir. 
Thanks!

**Comment by samuelimoisili** [comment]
The e2e test is failing on my computer. Here's the error I get each time I try to run it: 

```sh
(node:7554) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
 FAIL  test/ltfu-flow.spec.ts
  Steps to follow the Loss To Follow-Up (LTFU) workflow
    ✕ Should follow the LTFU workflow

  ● Steps to follow the Loss To Follow-Up (LTFU) workflow › Should follow the LTFU workflow

    Mediator channel installation failed

      43 |     }
      44 |   } catch (error) {
    > 45 |     throw new Error(`Mediator channel installation failed`);
         |           ^
      46 |   }
      47 | };
      48 | let placeId: string;

      at test/ltfu-flow.spec.ts:45:11
      at fulfilled (test/ltfu-flow.spec.ts:5:58)
```

**Comment by andrablaj** [comment]
Confirming that I get the same error as @samuelimoisili when running the e2e test locally:

Screenshot 2023-05-10 at 10 49 26

File: mediator/test/e2e-test.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+set -e
+BASEDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && cd .. && cd .. && pwd )"
+MEDIATORDIR="${BASEDIR}/mediator"
+DOCKERRDIR="${BASEDIR}/docker"
+
+
+export NODE_ENV=integration
+export NODE_TLS_REJECT_UNAUTHORIZED=0
```
Comment (andrablaj) [inline mediator/test/e2e-test.sh:9]: @samuelimoisili @lorerod, per [this comment ](https://stackoverflow.com/a/65021950/3541845), when changing this line to:

```
export NODE_TLS_REJECT_UNAUTHORIZED=1
```

in the development environment, solves the error that we were getting. 

However, now I get a different error related to the mediator installation:

Screenshot 2023-05-10 at 10 59 15

Comment (lorerod) [reply mediator/test/e2e-test.sh:9]: I will look into this today. Thanks for the feedback!

**Comment by lorerod** [comment]
@andrablaj I added the fixes you found when debugging. @samuelimoisili can you try again, please?

**Review by samuelimoisili — APPROVED** [review]

File: mediator/test/ltfu-flow.spec.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+const configureCHT = async () => {
+  const createPlaceResponse = await request(CHT.url)
+    .post('/api/v1/places')
+    .auth(CHT.username, CHT.password)
+    .send({ 'name': 'CHP Branch Two', 'type': 'district_hospital' });
+
+  if (createPlaceResponse.status === 200 && createPlaceResponse.body.ok === true) {
+    placeId = createPlaceResponse.body.id;
+  } else {
+    throw new Error(`CHT place creation failed: Reason ${createPlaceResponse.body}`);
```
[the remark this answers is not in the document]
Comment (lorerod) [reply mediator/test/ltfu-flow.spec.ts:62]: Thanks @njogz. Do you mean to use `createPlaceResponse.status` instead of `createPlaceResponse.body`?

**Comment by lorerod** [comment]
> Good job on this! I am still getting node error that @samuelimoisili mentioned.

Can you please try running `npm install` inside `./configurator` directory and then, running `npm install` and `npm run e2e-test` inside `./mediator` directory?

**Pull request merged**